waitress/app/tests/test_routes.py

```diff
@@ -208,6 +215,26 @@ def test_can_regularize_guest_names(self):
         regularized = regularize_guest_names(guest_list)
         assert regularized[-1].firstname == "Guest 10"
 
+    def test_user_can_view_admin_login_page(self):
+        response = self.client.get("")
+        assert response.status_code == 200
+        assert response.templates[0].name == "login.html"
+
+    def test_user_redirects_to_login_page_on_login_failure(self):
+        user_data = dict(username="user", password="password")
+        response = self.client.post("", data=user_data, follow=True)
+        assert response.status_code == 200
+        assert response.templates[0].name == "login.html"
+
+    def test_user_redirects_to_dashboard_page_on_login_success(self):
+        user_data = dict(username="username", password="userpassword")
+        response = self.client.post("", data=user_data, follow=True)
+        from nose.tools import set_trace
+
+        set_trace()
+        assert response.status_code == 200
+        assert response.templates[0].name == "dashboard.html"
```

Comment:
use `self.assertEqual` please

Reply:
fixed. Why is this a better alternative though?

Reply:
Already provided by the TestClass, has much more flexible api for deep equality and others you might think of. an example is effective testing of your templates without having to hardcode dashboard.html just search for what to expect